paperqa/docs.py

```diff
@@ -209,7 +209,15 @@ def set_client(
         client: Any | None = None,
         embedding_client: Any | None = None,
     ):
-        if client is None and isinstance(self.llm_model, OpenAILLMModel):
+        """Sets an API client for non LiteLLM compatibility.
+
+        If using the LiteLLMModel or LiteLLMEmbeddingModel, no client is needed.
+
+        """
+        if isinstance(self.llm_model, LiteLLMModel):
+            client = None
+        # below logic only for backwards compatibility
+        elif client is None and isinstance(self.llm_model, OpenAILLMModel):
```

Comment:
Let's break backwards compatibility and remove all this extra stuff

Reply:
@whitead what do you think?

Reply:
Yes, break it all

Reply:
it is broken